.github/workflows/release-check.yml

```diff
@@ -2,134 +2,165 @@ name: Release Checker
-        # If that is the case, we don't need to run the rest of the workflow.
-        if: env.VERSION != ''
+          echo "::set-output name=this::$version"
+      - id: tag_exists
+        name: Check if version tag already exists
+        if: steps.version.outputs.this != ''
         run: |
-          git fetch origin --tags
-          status=0
-          git rev-list $VERSION &> /dev/null || status=$?
-          if [[ $status != 0 ]]; then
-            echo "TAG_EXISTS=false" >> $GITHUB_ENV
-          fi
-      - name: Install semver (node command line tool)
-        if: env.TAG_EXISTS == 'false'
-        run: npm install -g "https://github.com/npm/node-semver#e79ac3a450e8bb504e78b8159e3efc7089569" # v7.3.5
+          # https://docs.github.com/en/rest/reference/git#get-a-reference
+          gh api '/repos/${{ github.repository }}/git/ref/tags/${{ steps.version.outputs.this }}' && tag_exists='true' || tag_exists='false'
+
+          echo "tag_exists=$tag_exists"
+          echo "::set-output name=this::$tag_exists"
+  check:
+    needs: [prepare]
+    if: needs.prepare.outputs.tag_exists != 'true'
+    runs-on: ubuntu-latest
+    env:
+      COMMENT: "Suggested version: `${{ needs.prepare.outputs.version }}`"
+      GITHUB_TOKEN: ${{ github.token }}
+    steps:
       - name: Check version
-        if: env.TAG_EXISTS == 'false'
-        run: semver ${{ env.VERSION }} # fails if the version is not a valid semver version (e.g. v0.1 would fail)
-      - name: Determine version number to compare to
-        if: env.TAG_EXISTS == 'false'
+        run: |
+          npm install -g 'https://github.com/npm/node-semver#e79ac3a450e8bb504e78b8159e3efc7089569' # v7.3.5
+
+          semver '${{ needs.prepare.outputs.version }}' # fails if the version is not a valid semver version (e.g. v0.1 would fail)
+      - id: release
+        name: Create draft GitHub Release (if it doesn't exist yet)
+        run: |
+          # using GraphQL because https://docs.github.com/en/rest/reference/releases#list-releases does not return draft releases
+          # https://docs.github.com/en/graphql/reference/objects#release
+          release="$(gh api graphql -f query='query { repository(owner: "${{ github.event.repository.owner.login }}", name: "${{ github.event.repository.name }}") { release(tagName: "${{ needs.prepare.outputs.version }}") { url } } }' --jq '.data.repository.release.url // ""')"
+          status=$? && ([[ $status == 0 ]] || exit $status)
+
+          if [[ -z "$release" ]]; then
+            # https://docs.github.com/en/rest/reference/releases#create-a-release
+            # creating a draft release does not create a tag, only publishing does
+            release="$(gh api '/repos/${{ github.repository }}/releases' -F 'draft=true' -f 'tag_name=${{ needs.prepare.outputs.version }}' -F 'generate_release_notes=true' --jq '.html_url')"
+            status=$? && ([[ $status == 0 ]] || exit $status)
+          fi
+
+          echo "COMMENT<<EOF
```

Comment:
TIL... cool pattern :)

Do we need to worry about script injection here?
https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#multiline-strings